### PR TITLE
Make top nav adjustments (size and highlight colour)

### LINKED ISSUE #4383: [New navigation] Top nav adjustments

### PR DESCRIPTION
**Pull request by bethcollins92**
## Done

- Update deprecated example to include more features
- Adjust the size of the top nav bar from `3.5rem` to `3rem` on small screens as per [design](https://www.figma.com/file/BSP7FqnRMcMvO7kCozJKUa/Ubuntu.com-navigation-update-Iteration-9?node-id=0%3A1832)
- Make the highlight the same as text instead of ubuntu orange

Fixes #4383 

## QA

- Open [demo](https://vanilla-framework-4386.demos.haus/)
- Go to the Navigation deprecated example and check they work as expected
- Check the highlight colour of the top nav is correct on light and dark themes
- Check on mobile the height of the top nav is correct (3rem)

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical-web-and-design/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical-web-and-design/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical-web-and-design/vanilla-framework/blob/main/templates/docs/whats-new.md).
- [x] Documentation side navigation should be updated with the relevant labels.

## Screenshots

Screenshot 2022-03-31 at 12 11 51
Screenshot 2022-03-31 at 12 12 16

### CODE REVIEW

**Comment by lyubomir-popov** [comment]
Looks great, thanks.

**Comment by bartaz** [comment]
Not sure why Percy is auto approved, there have been some small issues.

Deprecated example seems to be misaligned:
Screenshot 2022-04-04 at 10 20 35

First of all, this example should not have new logo (dark one).
Top example with old logo doesn't get smaller on mobile screens. "Menu" button is smaller but whole nav not (so button is misaligned).
This happens also in "Search box - Navigation" and "Search box - Navigation dark" examples, because they use current navigation component as well.

File: scss/_patterns_navigation.scss [hunk 1]
```diff
@@ -12,6 +13,11 @@ $spv-navigation-logo: 0.3125rem;
   %navigation-link-responsive-padding-vertical {
     padding-bottom: $spv--large;
     padding-top: $spv--large;
+
+    @media (max-width: $breakpoint-navigation-threshold) {
```
Comment (bartaz) [inline scss/_patterns_navigation.scss:17]: Ideally we should do this other way around (mobile first), so by default smaller spacing, and `min-width` breakpoint to turn it into desktop view.

File: scss/_patterns_navigation.scss [hunk 2]
```diff
@@ -208,6 +214,10 @@ $spv-navigation-logo: 0.3125rem;
       position: absolute;
       top: 0;
       width: $navigation-logo-tag-width;
+
+      @media (max-width: $breakpoint-navigation-threshold) {
```
Comment (bartaz) [inline scss/_patterns_navigation.scss:218]: Ideally we should do this other way around (mobile first), so by default smaller spacing, and `min-width` breakpoint to turn it into desktop view.

File: templates/docs/examples/patterns/navigation/deprecated.html [hunk 3]
```diff
@@ -32,36 +31,109 @@
...
   </div>
+  <div class="p-navigation__search-overlay"></div>
 </header>
 <header id="navigation" class="p-navigation is-dark">
   <div class="p-navigation__row">
     <div class="p-navigation__banner">
-      <div class="p-navigation__logo">
-        <a class="p-navigation__item" href="#">
-          <img class="p-navigation__image" src="https://assets.ubuntu.com/v1/3c7954dd-logo-canonical-white.svg" alt="" width="95" />
+      <div class="p-navigation__tagged-logo">
```
Comment (bartaz) [inline templates/docs/examples/patterns/navigation/deprecated.html:55]: Why are we using new logo in the "deprecated" navigation?

**Review by bartaz — CHANGES_REQUESTED** [review]
Some comments to address

**Pull request merged**